schemas/comorbidity.json

```diff
@@ -0,0 +1,97 @@
+{
+  "name": "comorbidity",
+  "description": "The collection of data elements related to a donor's comorbidities",
+  "meta": {
+     "parent": "donor"
+  },
+  "fields": [
+    {
+      "name": "program_id",
+      "valueType": "string",
+      "description": "Unique identifier of the ARGO program.",
+      "meta": {
+        "validationDependency": true,
+        "primaryId": true,
+        "foreignKey": "sample_registration.program_id",
+        "displayName": "Program ID"
+      },
+      "restrictions": {
+        "required": true
+      }
+    },
+    {
+      "name": "submitter_donor_id",
+      "valueType": "string",
+      "description": "Unique identifier of the donor, assigned by the data provider.",
+      "meta": {
+        "validationDependency": true,
+        "primaryId": true,
+        "foreignKey": "sample_registration.submitter_donor_id",
+        "displayName": "Submitter Donor ID"
+      },
+      "restrictions": {
+        "required": true,
+        "regex": "#/regex/submitter_id"
+      }
+    },
+    {
+      "name": "prior_malignancy",
+      "description": "Prior malignancy affecting donor.",
+      "restrictions": {
+        "codeList": "#/list/yes_no"
+      },
+      "valueType": "string",
+      "meta": {
+        "displayName": "Prior Malignancy"
+      }
+    },
+    {
+      "name": "laterality_of_prior_malignancy",
+      "description": "If donor has history of prior malignancy, indicate laterality of previous diagnosis. (Reference: caDSR CDE ID 4122391)",
+      "valueType": "string",
+      "restrictions": {
+        "codeList": ["Bilateral", "Left", "Midline", "Not applicable", "Right", "Unilateral, side not specified", "Unknown"],
+        "script": ""
+      },
+      "meta": {
+        "displayName": "Laterality at Prior Malignancy"
+      }
+    },
+    {
+      "name": "age_at_cormorbidity_diagnosis",
+      "valueType": "integer",
+      "description": "Indicate the age of comorbidity diagnosis, in years.",
+      "restrictions": {
+         "script": ""
+      },
+      "meta": {
+        "units": "years",
+        "displayName": "Age at Comorbidity Diagnosis"
+      }
+    },
+    {
+      "name": "comorbidity_type_code",
```

Comment:
Since multiple comorbidities can be submitted for a single donor we need to establish which record determines a comorbidity to be unique. 

suggestion to make this field required to accomplish this so the unique record value is:

so its `submitter_donor_id` + `comorbidity_type_code`

Reply:
Updated to make both primaryIds

Reply:
i think there is a problem here with this field not being `required` 

If its not required, then a user can submit 2 rows that look identical. for example a user includes these two rows:
```
`program id`, `submitter donor id`, `comorbidity treatment status`
DASH-CA, Donor-1, Yes 
DASH-CA, Donor- 1, Yes
```
In this  case, 1) we have 2 duplicate rows that are not identifiable and 2) is it a problem that we don't know what the comorbidity is?

Reply:
Good point @rosibaj. In addition to making `comorbidity_type_code` required as you suggested, I think `prior_malignancy` needs to be required too. This is because `comorbidity_type_code` is allowed to be empty if `prior_malignancy` is 'No' or 'Unknown'. But if `prior_malignancy` is optional, then it's difficult to enforce this requirement. I'll update the validation scripts to handle this as well.

Reply:
Good point @rosibaj. I changed `comorbidity_type_code` to required but attached a validation script to it to do the following:
- allow field to be empty if `prior_malignancy` is 'no' or 'unknown' (otherwise it is required).
- check submitted ICD-10 code. If `prior_malignancy` is 'yes', then the submitted code must be for a neoplasm (ie. starts with C or D). Otherwise, the submitted value in `comorbidity_type_code` just gets checked against the default regex specified in the dictionary.